cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -1648,6 +1661,60 @@ String zkConnectionString(int connectToReplicas)  {
                     });
         }
 
+        Future<ReconciliationState> kafkaBrokerDynamicConfiguration() {
+            KafkaBrokerConfigurationHelper kbch = new KafkaBrokerConfigurationHelper(kafkaSetOperations.getAdminClientProvider(), secretOperations);
+            return kafkaSetOperations.getAsync(namespace, KafkaCluster.kafkaClusterName(name))
+                .compose(sts -> {
+                    if (sts == null) {
```

Comment:
We just reconciled the STS right before runnign this. So I wonder if we really need to get it again from Kube. It seems to be later used only to extract namespace and cluster name which is IMHO something you can get much easier form the description and do not need to request the STS from Kube API server just because of it.